features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/TabCompositionGroupMapper.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.compose.internal.mappers
+
+import com.datadog.android.sessionreplay.compose.internal.data.Box
+import com.datadog.android.sessionreplay.compose.internal.data.ComposableParameter
+import com.datadog.android.sessionreplay.compose.internal.data.ComposeWireframe
+import com.datadog.android.sessionreplay.compose.internal.data.UiContext
+import com.datadog.android.sessionreplay.model.MobileSegment
+import com.datadog.android.sessionreplay.utils.ColorStringFormatter
+
+internal class TabCompositionGroupMapper(
+    colorStringFormatter: ColorStringFormatter
+) : AbstractCompositionGroupMapper(colorStringFormatter) {
+    override fun map(
+        stableGroupId: Long,
+        parameters: Sequence<ComposableParameter>,
+        boxWithDensity: Box,
+        uiContext: UiContext
+    ): ComposeWireframe {
+        var textColor: String? = uiContext.parentContentColor
+        val isSelected =
+            parameters.firstOrNull { it.name == "selected" }?.value as? Boolean ?: false
+        parameters.forEach { param ->
+            parseContentColor(isSelected, param)?.let {
+                textColor = it
+            }
+        }
+        return ComposeWireframe(
+            MobileSegment.Wireframe.ShapeWireframe(
+                id = stableGroupId,
+                x = boxWithDensity.x,
+                y = boxWithDensity.y,
+                width = boxWithDensity.width,
+                height = boxWithDensity.height
+            ),
+            uiContext.copy(parentContentColor = textColor)
+        )
+    }
+
+    private fun parseContentColor(isSelected: Boolean, param: ComposableParameter): String? {
```

Comment:
this looks a bit weird ...why not having 2 functions...one that returns the `selectedColor` and one the `unselectedColor` and based on the `isSelected` value you call one or the other ?

Reply:
you are right, updated